lib/mp-readline/readline.c

```diff
@@ -74,6 +74,7 @@ STATIC void mp_hal_move_cursor_back(uint pos) {
         // snprintf needs space for the terminating null character
         int n = snprintf(&vt100_command[0], sizeof(vt100_command), "\x1b[%u", pos);
         if (n > 0) {
+            assert(n < sizeof(vt100_command));
```

Comment:
This might have to go in a separate commit, plus it's not going to compile with all warnings enabled because of comparision between signed/unsigned.

Reply:
I'll move to a separate commit.

It compiles with asserts enabled, I've used it to discover one of the bugs (where `pos` would underflow and this function will actually receive `2 ** 32 - 1`). The assert was raised in that case.

Reply:
> It compiles with asserts enabled

Well it indeed doesn't compile on the coverage build! I've added a cast to `unsigned`.
It's funny we get this warning because it's in an `n > 0` block, so I don't see any signed/unsigned conversion problem that might occur.

Reply:
> because it's in an n > 0 block

hehe yes, but compilers do not follow that reasoning, luckily